src/app/static/src/tests/components/downloadResults/uploadModal.test.ts

```diff
@@ -490,4 +490,32 @@ describe(`uploadModal `, () => {
         expect(labels.at(1).attributes("for")).toBe("id-0-1");
         expect(labels.at(2).attributes("for")).toBe("id-1-0");
     });
+
+    it(`does not send upload files to ADR if files are being prepared for upload`, async () => {
```

Comment:
This looks like it tests the changed button behaviour... but not the change made to address the underlying issue i.e. the fix to `confirmUpload`?

So doesn't appear to offer assurance that that issue is resolved.